### PR TITLE
better approach for game init

### PR DESCRIPTION
**Pull request by time-killer-games**
Originally I was firing window_center() for 15 steps in a row, to accommodate for the fact, it would not center regardless sometimes. This time, I changed it so that it while loops until it is centered.

### CODE REVIEW

**Comment by time-killer-games** [comment]
@JoshDreamland Ok, this one's ready for review! Thanks!

**Comment by time-killer-games** [comment]
@JoshDreamland yeah i tested this and it doesn't break views or anything, so it should be ready for review. I will say that I noticed even those I have window_default()'s center_size argument set to false, it still centers the game window at the start of every room, and I'm not sure how to make it stop doing that.

**Comment by time-killer-games** [comment]
@JoshDreamland ready for review, thanks!

File: ENIGMAsystem/SHELL/Platforms/General/PFwindow.cpp [hunk 1]
```diff
@@ -287,8 +287,9 @@ void window_default(bool center_size) {
 }
 
 void window_center() {
-  window_set_position(display_get_width() / 2 - window_get_width() / 2,
-                      display_get_height() / 2 - window_get_height() / 2);
+  // without this while loop, the window won't be guaranteed to center at game init, in xlib, which is weird, but this makes it necessary.
+  while (enigma::windowX != display_get_width() / 2 - window_get_width() / 2 || enigma::windowY != display_get_height() / 2 - window_get_height() / 2)
```
Comment (JoshDreamland) [inline ENIGMAsystem/SHELL/Platforms/General/PFwindow.cpp]: Why is this a while loop?

Comment (time-killer-games) [reply ENIGMAsystem/SHELL/Platforms/General/PFwindow.cpp]: What do you mean 'bound this'?

Comment (time-killer-games) [reply ENIGMAsystem/SHELL/Platforms/General/PFwindow.cpp]: On a similar topic, you never answered my question on discord if this for loop might loop forever on titling window managers and other similar circumstances, I might as well address that in this pr as well if you're game: https://github.com/enigma-dev/enigma-dev/blob/c9352a7e1a52b0add48543ed76c01ad12bca9c83/ENIGMAsystem/SHELL/Platforms/xlib/XLIBwindow.cpp#L325

Comment (time-killer-games) [reply ENIGMAsystem/SHELL/Platforms/General/PFwindow.cpp]: Alright, I took care of it. I'll let you know when it passes. Thanks!

**Comment by time-killer-games** [comment]
@JoshDreamland Alright, all checks have passed.

**Pull request merged**